spla 1.6.0 doesn't compile with rocblas 6.0.2

```
     210    spla/src/gpu_util/gpu_runtime_api.hpp:258:13: note: in call to 'spla::gpu::StatusType spla::gpu::get_last_error()', declared h
            ere
     211      258 | inline auto get_last_error() -> StatusType { return GPU_PREFIX(GetLastError)(); }
     212          |             ^~~~~~~~~~~~~~
     213    spla/src/gpu_util/gpu_runtime_api.hpp:52:7: note: 'spla::gpu::StatusType' {aka 'hipError_t'} declared here
     214       52 | using StatusType = GPU_PREFIX(Error_t);
     215          |       ^~~~~~~~~~
  >> 216    spla/src/gpu_util/gpu_pointer_translation.hpp:60:26: error: 'using spla::gpu::PointerAttributes = hipPointerAttribute_t' {aka 
            'struct hipPointerAttribute_t'} has no member named 'memoryType'
     217       60 |   auto memoryType = attr.memoryType;
     218          |                          ^~~~~~~~~~
```

where `hipPointerAttribute_t` (from `/opt/rocm/include/hip/hip_runtime_api.h`) is:
```
/**
 * Pointer attributes
 */
typedef struct hipPointerAttribute_t {
    enum hipMemoryType type;
    int device;
    void* devicePointer;
    void* hostPointer;
    int isManaged;
    unsigned allocationFlags; /* flags specified when memory was allocated*/
    /* peers? */
} hipPointerAttribute_t;
``` 
Seems `memoryType` has been renamed to `type`?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -119,7 +119,7 @@ jobs:
         cd ${HOME}
         wget https://repo.radeon.com/rocm/rocm.gpg.key
         cat rocm.gpg.key  | gpg --dearmor | tee /etc/apt/keyrings/rocm.gpg > /dev/null
-        echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/${{ rocm_version }} jammy main" | tee --append /etc/apt/sources.list.d/rocm.list
+        echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/${{ matrix.rocm_version }} jammy main" | tee --append /etc/apt/sources.list.d/rocm.list
         apt-get update
         apt-get install -y hip-dev rocblas-dev
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -104,7 +104,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        rocm_version: [5.6, 6.1.2]
+        rocm_version: [5.6, 6.0.2]
 
     runs-on: ubuntu-22.04
     container: rocm/dev-ubuntu-22.04:${{ matrix.rocm_version }}-complete

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -107,7 +107,6 @@ jobs:
         rocm_version: [5.6, 6.0.2]
 
     runs-on: ubuntu-22.04
-    container: rocm/dev-ubuntu-22.04:${{ matrix.rocm_version }}
 
     steps:
     # Checks-out your repository under $GITHUB_WORKSPACE
@@ -116,7 +115,14 @@ jobs:
     - name: Install dependencies
       run: |
         apt-get update
-        DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC  apt-get install -y make g++ libopenblas-dev libscalapack-mpi-dev wget git make cmake
+        DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC  apt-get install -y make g++ libopenblas-dev libscalapack-mpi-dev wget git make cmake gpg
+        cd ${HOME}
+        wget https://repo.radeon.com/rocm/rocm.gpg.key
+        cat rocm.gpg.key  | gpg --dearmor | tee /etc/apt/keyrings/rocm.gpg > /dev/null
+        echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/${{ rocm_version }} jammy main" | tee --append /etc/apt/sources.list.d/rocm.list
+        apt-get update
+        apt-get install -y hip-dev rocblas-dev
+
 
     - name: Build
       run: |

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -107,7 +107,7 @@ jobs:
         rocm_version: [5.6, 6.0.2]
 
     runs-on: ubuntu-22.04
-    container: rocm/dev-ubuntu-22.04:${{ matrix.rocm_version }}-complete
+    container: rocm/dev-ubuntu-22.04:${{ matrix.rocm_version }}
 
     steps:
     # Checks-out your repository under $GITHUB_WORKSPACE

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -121,7 +121,7 @@ jobs:
         cat rocm.gpg.key  | gpg --dearmor | sudo tee /etc/apt/keyrings/rocm.gpg > /dev/null
         echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/${{ matrix.rocm_version }} jammy main" | sudo tee --append /etc/apt/sources.list.d/rocm.list
         sudo apt-get update
-        sudo apt-get install -y hip-dev rocblas-dev
+        sudo apt-get install -y hip-dev rocblas-dev rocm-device-libs
 
 
     - name: Build

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -100,7 +100,7 @@ jobs:
   #################
   # Build with ROCm
   #################
-  build-rocm-5.6:
+  build-rocm:
     strategy:
       fail-fast: false
       matrix:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -114,14 +114,14 @@ jobs:
 
     - name: Install dependencies
       run: |
-        apt-get update
-        DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC  apt-get install -y make g++ libopenblas-dev libscalapack-mpi-dev wget git make cmake gpg
+        sudo apt-get update
+        DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC sudo apt-get install -y make g++ libopenblas-dev libscalapack-mpi-dev wget git make cmake gpg
         cd ${HOME}
         wget https://repo.radeon.com/rocm/rocm.gpg.key
-        cat rocm.gpg.key  | gpg --dearmor | tee /etc/apt/keyrings/rocm.gpg > /dev/null
-        echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/${{ matrix.rocm_version }} jammy main" | tee --append /etc/apt/sources.list.d/rocm.list
-        apt-get update
-        apt-get install -y hip-dev rocblas-dev
+        cat rocm.gpg.key  | gpg --dearmor | sudo tee /etc/apt/keyrings/rocm.gpg > /dev/null
+        echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/${{ matrix.rocm_version }} jammy main" | sudo tee --append /etc/apt/sources.list.d/rocm.list
+        sudo apt-get update
+        sudo apt-get install -y hip-dev rocblas-dev
 
 
     - name: Build

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -100,9 +100,14 @@ jobs:
   #################
   # Build with ROCm
   #################
-  build-rocm:
+  build-rocm-5.6:
+    strategy:
+      fail-fast: false
+      matrix:
+        rocm_version: [5.6, 6.1.2]
+
     runs-on: ubuntu-22.04
-    container: rocm/dev-ubuntu-22.04:5.6-complete
+    container: rocm/dev-ubuntu-22.04:${{ matrix.rocm_version }}-complete
 
     steps:
     # Checks-out your repository under $GITHUB_WORKSPACE

--- a/src/gpu_util/gpu_blas_handle.hpp
+++ b/src/gpu_util/gpu_blas_handle.hpp
@@ -32,6 +32,7 @@
 #if defined(SPLA_CUDA) || defined(SPLA_ROCM)
 #include <cassert>
 #include <memory>
+#include <tuple>
 
 #include "gpu_util/gpu_blas_api.hpp"
 #include "gpu_util/gpu_runtime_api.hpp"
@@ -47,7 +48,7 @@ public:
 
     handle_ = std::shared_ptr<gpu::blas::HandleType>(new gpu::blas::HandleType(rawHandle),
                                                      [](gpu::blas::HandleType* ptr) {
-                                                       gpu::blas::destroy(*ptr);
+                                                       std::ignore = gpu::blas::destroy(*ptr);
                                                        delete ptr;
                                                      });
 

--- a/src/gpu_util/gpu_device_guard.hpp
+++ b/src/gpu_util/gpu_device_guard.hpp
@@ -29,6 +29,7 @@
 #define SPLA_GPU_DEVICE_GUARD_HPP
 
 #include <memory>
+#include <tuple>
 
 #include "gpu_util/gpu_runtime_api.hpp"
 #include "spla/config.h"
@@ -51,7 +52,8 @@ public:
 
   ~GPUDeviceGuard() {
     if (targetDeviceId_ != originalDeviceId_) {
-      gpu::set_device(originalDeviceId_);  // no check to avoid throw exeception in destructor
+      // no check to avoid throw exeception in destructor
+      std::ignore = gpu::set_device(originalDeviceId_);
     }
   }
 

--- a/src/gpu_util/gpu_event_handle.hpp
+++ b/src/gpu_util/gpu_event_handle.hpp
@@ -31,6 +31,7 @@
 #include "spla/config.h"
 #if defined(SPLA_CUDA) || defined(SPLA_ROCM)
 #include <memory>
+#include <tuple>
 
 #include "gpu_util/gpu_runtime_api.hpp"
 #include "spla/exceptions.hpp"
@@ -46,7 +47,7 @@ public:
     gpu::check_status(gpu::event_create_with_flags(&event, flag));
 
     event_ = std::shared_ptr<gpu::EventType>(new gpu::EventType(event), [](gpu::EventType* ptr) {
-      gpu::event_destroy(*ptr);
+        std::ignore = gpu::event_destroy(*ptr);
       delete ptr;
     });
   };

--- a/src/gpu_util/gpu_pointer_translation.hpp
+++ b/src/gpu_util/gpu_pointer_translation.hpp
@@ -29,6 +29,7 @@
 #ifndef SPLA_GPU_POINTER_TRANSLATION_HPP
 #define SPLA_GPU_POINTER_TRANSLATION_HPP
 
+#include <tuple>
 #include <utility>
 
 #include "gpu_util/gpu_runtime_api.hpp"
@@ -45,7 +46,7 @@ auto translate_gpu_pointer(const T* inputPointer) -> std::pair<const T*, const T
   auto status = gpu::pointer_get_attributes(&attr, static_cast<const void*>(inputPointer));
 
   if (status != gpu::status::Success) {
-    gpu::get_last_error();  // clear error from cache
+    std::ignore = gpu::get_last_error();  // clear error from cache
     // Invalid value is always indicated before CUDA 11 for valid host pointers, which have not been
     // registered. -> Don't throw error in this case.
     if (status != gpu::status::ErrorInvalidValue) gpu::check_status(status);

--- a/src/gpu_util/gpu_pointer_translation.hpp
+++ b/src/gpu_util/gpu_pointer_translation.hpp
@@ -53,8 +53,8 @@ auto translate_gpu_pointer(const T* inputPointer) -> std::pair<const T*, const T
 
   std::pair<const T*, const T*> ptrPair{nullptr, nullptr};
 
-  // get memory type - cuda 10 changed attribute name
-#if defined(SPLA_CUDA) && (CUDART_VERSION >= 10000)
+  // get memory type - cuda 10 and HIP 6.0 changed attribute name
+#if (defined(SPLA_CUDA) && (CUDART_VERSION >= 10000)) || (defined(SPLA_ROCM) && (HIP_VERSION_MAJOR >= 6))
   auto memoryType = attr.type;
 #else
   auto memoryType = attr.memoryType;

--- a/src/gpu_util/gpu_stream_handle.hpp
+++ b/src/gpu_util/gpu_stream_handle.hpp
@@ -31,6 +31,7 @@
 #include "spla/config.h"
 #if defined(SPLA_CUDA) || defined(SPLA_ROCM)
 #include <memory>
+#include <tuple>
 
 #include "gpu_util/gpu_runtime_api.hpp"
 #include "spla/exceptions.hpp"
@@ -49,7 +50,7 @@ public:
 
     stream_ =
         std::shared_ptr<gpu::StreamType>(new gpu::StreamType(rawStream), [](gpu::StreamType* ptr) {
-          gpu::stream_destroy(*ptr);
+          std::ignore = gpu::stream_destroy(*ptr);
           delete ptr;
         });
   };

--- a/src/memory/allocator_collection.hpp
+++ b/src/memory/allocator_collection.hpp
@@ -37,6 +37,7 @@
 #include <cstdlib>
 #include <functional>
 #include <memory>
+#include <tuple>
 
 #include "memory/allocator.hpp"
 #include "memory/pool_allocator.hpp"
@@ -70,14 +71,14 @@ public:
               gpu::check_status(gpu::malloc_host(&ptr, size));
               return ptr;
             },
-            [](void* ptr) -> void { gpu::free_host(ptr); })),
+            [](void* ptr) -> void { std::ignore = gpu::free_host(ptr); })),
         allocGPU_(new PoolAllocator<MemLoc::GPU>(
             [](std::size_t size) -> void* {
               void* ptr = nullptr;
               gpu::check_status(gpu::malloc(&ptr, size));
               return ptr;
             },
-            [](void* ptr) -> void { gpu::free(ptr); }))
+            [](void* ptr) -> void { std::ignore = gpu::free(ptr); }))
 #endif
   {
     int mpiInitialized = 0;

--- a/src/pgemm_sbs/ring_sbs_gpu.cpp
+++ b/src/pgemm_sbs/ring_sbs_gpu.cpp
@@ -284,13 +284,13 @@ auto RingSBSGPU<T, BLOCK_GEN>::process_step_ring(std::unordered_set<IntType>& be
       // Make sure no other stream is writing to the same location. Select event based on coloumn
       // index, which determines the write location.
       auto& event = colEvents[(block.col / block.numCols) % colEvents.size()];
-      gpu::stream_wait_event(proc.blasHandle.stream_handle().get(), event.get(), 0);
+      gpu::check_status(gpu::stream_wait_event(proc.blasHandle.stream_handle().get(), event.get(), 0));
 
       sbs_gemm_gpu_async<T>(proc.blasHandle, alpha_,
                             proc.matA.sub_accessor(0, block.row, proc.matA.rows(), block.numRows),
                             GPUConstMatrixAccessor<T>(blockViewGPU), beta,
                             proc.matC.sub_accessor(0, block.col, proc.matC.rows(), block.numCols));
-      gpu::event_record(event.get(), proc.blasHandle.stream_handle().get());
+      gpu::check_status(gpu::event_record(event.get(), proc.blasHandle.stream_handle().get()));
     }
   }
   state_ = stepIdx_ >= comm_.size() - 1 ? TileState::Empty : TileState::PartiallyProcessed;
@@ -337,13 +337,13 @@ auto RingSBSGPU<T, BLOCK_GEN>::process_step_broadcast(std::unordered_set<IntType
       // Make sure no other stream is writing to the same location. Select event based on coloumn
       // index, which determines the write location.
       auto& event = colEvents[(block.col / block.numCols) % colEvents.size()];
-      gpu::stream_wait_event(proc.blasHandle.stream_handle().get(), event.get(), 0);
+      gpu::check_status(gpu::stream_wait_event(proc.blasHandle.stream_handle().get(), event.get(), 0));
 
       sbs_gemm_gpu_async<T>(proc.blasHandle, alpha_,
                             proc.matA.sub_accessor(0, block.row, proc.matA.rows(), block.numRows),
                             GPUConstMatrixAccessor<T>(blockViewGPU), beta,
                             proc.matC.sub_accessor(0, block.col, proc.matC.rows(), block.numCols));
-      gpu::event_record(event.get(), proc.blasHandle.stream_handle().get());
+      gpu::check_status(gpu::event_record(event.get(), proc.blasHandle.stream_handle().get()));
     }
   }
 

